GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -320,7 +321,29 @@ string ComposeOutput()
 
             // Wait for the process to exit (or be cancelled) and for the output
             Task<int> exitTask = process.WaitForExitAsync(cancellationToken);
-            await Task.WhenAll(outputTask, errorTask, exitTask);
+            try
+            {
+                await Task.WhenAll(outputTask, errorTask, exitTask);
+            }
+            catch (ExternalOperationException ex)
+            {
+                string errorTxt = null;
+                try
+                {
+                    errorTxt = outputEncoding.GetString(errorBuffer.GetBuffer(), 0, (int)errorBuffer.Length);
+                }
+                catch (Exception)
+                {
+                    // We want to ensure that this action won't crash
+                }
+
+                if (errorTxt?.Contains(DubiousOwnershipSecurityConfigString) is true)
+                {
+                    throw new ExternalOperationException(ex.Command, ex.Arguments, ex.WorkingDirectory, ex.ExitCode, new Exception(errorTxt));
+                }
```

Comment:
```suggestion
                try
                {
                    string errorText = outputEncoding.GetString(errorBuffer.GetBuffer(), 0, (int)errorBuffer.Length);
                    if (errorText.Contains(DubiousOwnershipSecurityConfigString))
                    {
                        throw new ExternalOperationException(ex.Command, ex.Arguments, ex.WorkingDirectory, ex.ExitCode, new Exception(errorText));
                    }
                }
                catch (Exception)
                {
                    // We want to ensure that this action won't crash
                }

```

Reply:
Sorry but that's not what we want to do.
Putting the throw inside the try will catch the exception whereas we want to propagate it with an inner exception containing this time the security issue text and that will be well handled by the bug reporter.

Reply:
Of course! What was I thinking 🤔